compiler/rustc_const_eval/src/interpret/operand.rs

```diff
@@ -569,8 +569,10 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             ty::ConstKind::Unevaluated(uv) => {
                 let instance = self.resolve(uv.def, uv.substs)?;
                 let cid = GlobalId { instance, promoted: None };
-                self.ctfe_query(span, |tcx| tcx.eval_to_valtree(self.param_env.and(cid)))?
-                    .unwrap_or_else(|| bug!("unable to create ValTree for {uv:?}"))
+                self.ctfe_query(span, |tcx| {
+                    tcx.eval_to_valtree(self.param_env.with_const().and(cid))
```

Comment:
I think the eval_to_valtree query is just missing a `remap_constness`https://github.com/rust-lang/rust/blob/cbc70ff277dda8b7f227208eff789f1f68b6de5a/compiler/rustc_middle/src/query/mod.rs#L1013

I'll be so happy when the constness field is finally gone

Reply:
@oli-obk doesn't `remap_env_constness` set the param-env constness to *not const*?

Also, none of the other eval-like queries use `remap_env_constness`.